numpyro/distributions/continuous.py

```diff
@@ -1320,3 +1320,118 @@ def tree_flatten(self):
     @classmethod
     def tree_unflatten(cls, aux_data, params):
         return cls(batch_shape=aux_data)
+
+
+class ProjectedNormal(TorchDistribution):
```

Comment:
Probably `directional.py` is a better place for this distribution. Just thinking about this today, this is a nice distribution to have and the approach here looks very reasonable to me.

Just a note for myself: tell users to use this distribution and reparam (probably registering `biject_to` for `sphere`, but raise an error there) instead of using ImproperUniform. `concentration=0` means uniform over the sphere. No need to have `scale` parameter because, under projection, the location of `concentration` is enough to determine the distribution over sphere.